plutus-contract/src/Language/Plutus/Contract/Test.hs

```diff
@@ -207,10 +317,20 @@ assertRecord w p nm = PredF $ \(_, rs) ->
             tellSeq ["Record failed with", show err, "in '" <> nm <> "'"]
             pure False
 
-assertResult :: Wallet -> (Maybe a -> Bool) -> String -> TracePredicate a
+assertResult
+    :: forall ρ σ a. 
+       ( Forall ρ Show
+       , AllUniqueLabels σ
+       , Forall σ Semigroup
+       , Forall σ Monoid
```

Comment:
Having seen this a bunch of times, you can probably save yourself some ink by omitting `Forall sigma Semigroup` since presumably it's implied by `Forall sigma Monoid`.

Reply:
> `Forall sigma Semigroup` since presumably it's implied by `Forall sigma Monoid`

`Forall` is a type class, so I doubt it.